### DISCUSSION TITLE
Behavior of [link] terms spanning 3 residues

### DISCUSSION THREAD

**Discussion by ricalessandri** [marrink-lab/polyply_1.0#218] [Q&A] [answered]
Gonna try the discussions :)

Weird behavior - at least for me - observed when having _**links spanning 3 residues**_. Namely, seemingly equivalent lines would give different behaviors.

Namely, the following 2 combinations would work as expected, i.e., print those "i-j-k" terms when needed (i.e., at least 3 residues present):
```
[link]
resname "PMMA"
[ dihedrals ]
++C01  +C02  +C01   C02     3      2.301     -1.464      0.837     -1.674     -0.000      0.000   {"comment":"i-j-k", "group":"connection"}
[ pairs ]
  C02  ++C01       1   {"comment":"i-j-k", "group":"connection"}
```
  ```
[link]
resname "PMMA"
[ dihedrals ]
 +C01   C02   C01  -C02     3      2.301     -1.464      0.837     -1.674     -0.000      0.000   {"comment":"i-j-k", "group":"connection"}
[ pairs ]
 -C02   +C01       1   {"comment":"i-j-k", "group":"connection"}
```
 
While the following combinations will **not** print these terms in the itp:
```
[link]
resname "PMMA"
[ dihedrals ]
 +C01   C02   C01  -C02     3      2.301     -1.464      0.837     -1.674     -0.000      0.000   {"comment":"i-j-k", "group":"connection"}
[ pairs ]
  C02  ++C01       1   {"comment":"i-j-k", "group":"connection"}
```
```
[link]
resname "PMMA"
[ dihedrals ]
 ++C01  +C02  +C01   C02     3      2.301     -1.464      0.837     -1.674     -0.000      0.000   {"comment":"i-j-k", "group":"connection"}
[ pairs ]
 -C02   +C01       1   {"comment":"i-j-k", "group":"connection"}
```
or maybe I'm missing something and this is intended (for some good reason)?

**Comment by fgrunewald** [comment, marked as answer]
@ricalessandri the problem here is the pair directive which in the last two examples has a prefix that does not occur in dihedrals so it actually requires 5 residues not 4

**Reply by ricalessandri** [reply]
@fgrunewald  you mean 4 residues instead of 3, right? (`-` = residue 1, ` ` = residue 2, `+` = residue 3, `++` = residue 4)

OK, so that means that a link cannot span more than 3 residues?

**Reply by fgrunewald** [reply]
My bad my math was off: Case 1 & 2 span 3 residues, but case 3 & 4 span 4 residues, so they are not equivalent.  Plus pairs don't make edges, which means that you have a disconnected residue in case 3&4. You need to edges between the correct atoms in the link.

**Comment by fgrunewald** [comment]
Nice use of the discussion feature by the way!
